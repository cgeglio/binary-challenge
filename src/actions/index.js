export const login = user => ({
  type: 'LOGIN',
  user
})

export const logout = user => ({
  type: 'LOGOUT',
  user
})

export const addCards = cards => ({
  type: 'ADD_CARDS',
  cards
})

export const removeCards = cards => ({
  type: 'REMOVE_CARDS',
  cards
})

export const addQuestion = question => ({
  type: 'ADD_QUESTION',
  question
})

export const removeQuestion = question => ({
  type: 'REMOVE_QUESTION',
  question
})

export const addFortune = fortune => ({
  type: 'ADD_FORTUNE',
  fortune
})

export const removeFortune = fortune => ({
  type: 'REMOVE_FORTUNE',
  fortune
})

export const addReading = currentReading => ({
  type: 'ADD_READING',
  currentReading
})

export const removeReading = currentReading => ({
  type: 'REMOVE_READING',
  currentReading
})

export const addFavorite = favorite => ({
  type: 'ADD_FAVORITE',
  favorite
})

export const removeFavorite = favorite => ({
  type: 'REMOVE_FAVORITE',
  favorite
})

export const resetFavorites = favorites => ({
  type: 'RESET_FAVORITES',
  favorites
})
