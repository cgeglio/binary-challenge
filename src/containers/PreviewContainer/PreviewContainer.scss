@import "../../variables.scss";
@import "../../mixins.scss";

.preview-container {
  @include flex();
  background-color: black;
  height: auto;
}

.saved-title {
  font-size: 1.5em;
  margin-bottom: 30px;
}

.saved-error {
  color: $primary-color;
}
