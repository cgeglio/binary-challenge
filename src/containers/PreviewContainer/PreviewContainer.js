import React, { Component } from 'react';
import './PreviewContainer.scss';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { removeQuestion } from '../../actions';
import ReadingPreview from '../../components/ReadingPreview/ReadingPreview';
import PropTypes from 'prop-types';

export class PreviewContainer extends Component {

  resetQuestion = () => {
    this.props.resetQuestionInStore(this.props.question);
  }

  render() {
    return (
      <section className='preview-container fade-in'>
        <h2 className='saved-title'>Saved Readings</h2>
        {!this.props.favorites.length ? <h2 className='saved-error'>You have not saved any readings yet.</h2> :
        this.props.favorites.map(favorite => {
          return <ReadingPreview key={favorite.id} reading={favorite} />
        })
        }
        <Link to='/home'><button onClick={() => this.resetQuestion()} className='back-btn'>Home</button></Link>
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  favorites: state.favorites,
  question: state.question,
})

export const mapDispatchToProps = dispatch => ({
  resetQuestionInStore: question => (dispatch(removeQuestion(question)))
})

export default connect(mapStateToProps, mapDispatchToProps)(PreviewContainer);

PreviewContainer.propTypes = {
  favorites: PropTypes.array,
  question: PropTypes.string,
  resetQuestionInStore: PropTypes.func
}
