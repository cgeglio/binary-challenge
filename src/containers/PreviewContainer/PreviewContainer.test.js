import React from 'react';
import { PreviewContainer, mapStateToProps, mapDispatchToProps } from './PreviewContainer';
import { removeQuestion } from '../../actions';
import { shallow } from 'enzyme';

describe('PreviewContainer', () => {

  describe('PreviewContainer container/component', () => {
    it ('should match the snapshot', () => {
      const mockFavorites = [{cards: {name: 'The Magician'}, id: 20}]
      const wrapper = shallow(<PreviewContainer favorites={mockFavorites} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an array of favorites and a question', () => {
      const mockDispatch = jest.fn();
      const mockFavorites = [{cards: [{name: 'The Magician'}], id: 20}];
      const wrapper = shallow(<PreviewContainer favorites={mockFavorites} />);
      const mockState = {
        user: {user: {name: 'Bithcuits', id: 9} },
        question: 'Is the sky blue?',
        favorites: [
          {cards: [{name: 'The Magician'}], id: 20},
          {cards: [{name: 'The Moon'}], id: 10}
        ]
      };
      const expected = {
        question: 'Is the sky blue?',
        favorites: [
          {cards: [{name: 'The Magician'}], id: 20},
          {cards: [{name: 'The Moon'}], id: 10}
      ]};
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    })
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the removeQuestion action when resetQuestion is called', () => {
      let mockResetQuestion = jest.fn().mockImplementation();
      let mockFavorites = [
        {cards: [{name: 'The Magician'}], id: 20},
        {cards: [{name: 'The Moon'}], id: 10}
      ];
      const wrapper = shallow(<PreviewContainer favorites={mockFavorites} removeQuestion={mockResetQuestion} />);
      const mockDispatch = jest.fn();
      const actionToDispatch = removeQuestion('question');
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.resetQuestion('question');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});
