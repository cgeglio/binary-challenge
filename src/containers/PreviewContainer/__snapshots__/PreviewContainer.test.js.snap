// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewContainer PreviewContainer container/component should match the snapshot 1`] = `
<section
  className="preview-container fade-in"
>
  <h2
    className="saved-title"
  >
    Saved Readings
  </h2>
  <Connect(ReadingPreview)
    key="20"
    reading={
      Object {
        "cards": Object {
          "name": "The Magician",
        },
        "id": 20,
      }
    }
  />
  <Link
    to="/home"
  >
    <button
      className="back-btn"
      onClick={[Function]}
    >
      Home
    </button>
  </Link>
</section>
`;
